### PR TITLE
Add required directories for actions and update-deps action

### PR DESCRIPTION
**Pull request by skillitzimberg**
## Description

@stacietaylorcima had requested a PR to automate updating of new team repos so that mentors don't have to run `npm update` and push any changes manually.

This PR introduces a GitHub action that runs `npm update` in a new repo that is generated from this template repo.

This workflow has one job - `npm-update`. It will only run when the new repo's `main` branch is created.

This is what happens when this workflow runs:  
- GitHub creates a virtual machine and installs `macos`.
- Another action from the marketplace called `checkout` runs which checks out the newly generated repo into the virtual machine.
- The marketplace action `setup-node` installs Node.
- `npm update` is run.
- `.github` is untracked from Git and removed from the repo.
- The deletion of `.github` and any updates to `package-lock.json` are added and committed.
- The changes are pushed from the virtual machine back to the `main` branch of the new repo.

The `.github` directory is removed from the new repo because it has nothing to do with what the team is working on so it's just noise.

@mxmason has a PR open and approved that moves some dependencies into devDependencies. If that is merged into `main` the `--dev` flag should be added to the workflow's `npm update` command so that those dependencies are included in any updates that may take place.

## Testing Steps / QA Criteria

I'm not totally sure of the best way for reviewers to test this. I think that forking this repo and testing the creation of new repos from the template may be the simplest.

1. Fork `https://github.com/skillitzimberg/smart-shopping-list` to your GitHub account.
2. From your fork, click the `Use This Template` button.
3. Name the new repo you are generating from the template.
4. Click the `Create repository from template` button.
5. Once the new repo is generated, go to the `Actions` tab.
6. To view logs from the steps being run in the action, click the `update-deps` workflow.

Sometimes `npm update` can take a while, just like it can on your own computer and the GitHub ui can hang up. Just refresh the page or come back to it in a few minutes.

### CODE REVIEW

File: .github/workflows/update-deps.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: update-deps
+on: create
+jobs:
+  npm-update:
+    if: github.ref == 'refs/heads/main'
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - run: npm update
```
Comment (segdeha) [inline .github/workflows/update-deps.yml]: Is there any harm in adding the `--dev` flag here before @mxmason’s PR is merged?

File: .github/workflows/update-deps.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: update-deps
+on: create
+jobs:
+  npm-update:
+    if: github.ref == 'refs/heads/main'
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (segdeha) [inline .github/workflows/update-deps.yml:8]: What’s the `@v2` here mean? Will we need to update this at some point to `@v3`?

File: .github/workflows/update-deps.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      - uses: actions/setup-node@v2
+      - run: npm update
+      - run: git rm -r --cached .github
+      - run: rm -rf .github/
+      - uses: EndBug/add-and-commit@v7.0.0
```
Comment (segdeha) [inline .github/workflows/update-deps.yml]: The version numbers here have me wondering if there’s something we should track in terms of when these should be updated? Or is it possible to use `npm` style `^` or something else to allow patches without having to explicitly change the number except for major versions?

**Review by segdeha — APPROVED** [review]
This is so great, @skillitzimberg! I have a couple of questions in the comments, but I don’t see anything that should prevent us from merging this. Nice work!!!

**Comment by skillitzimberg** [comment]
I decided that the GH Actions was the best way to handle this because while Dependabot can keep the deps up-to-date, it will create PRs that someone will have to go through and merge. Or you can set up a GH Action to automate the merging of those PRs, so this solution just seemed simpler.

File: .github/workflows/update-deps.yml (hunk 1 above)
Comment (skillitzimberg) [reply .github/workflows/update-deps.yml]: I don't think so. It just wasn't directly related to the way things are currently set up so I left it alone.

File: .github/workflows/update-deps.yml (hunk 2 above)
Comment (skillitzimberg) [reply .github/workflows/update-deps.yml:8]: Good question! [Here's the repo](https://github.com/actions/checkout). `v2` came out Dec 12, 2019. They got to `v2.3.4` in Nov. 2020. At some point I'm sure it'll have to happen, but there doesn't seem to be a whole lot of churn.

File: .github/workflows/update-deps.yml (hunk 3 above)
Comment (skillitzimberg) [reply .github/workflows/update-deps.yml]: There are some marketplace actions that will let you pull the latest version of a repo. But that seems like chasing your tail as you'd have to update that as well.

I'll look into this. Nothing comes up for me immediately, so I'll have to poke around.

Comment (skillitzimberg) [reply .github/workflows/update-deps.yml]: It is possible to use `action-name@latest`, but only if the maintainers are tagging a `latest` branch. `EndBug` actually has an action to help maintainers with this. `EndBug/add-and-commit@latest` is `7.0.0`. I'll update this. `checkout` and `setup-node` are not using the `latest` tag so those would need to be monitored.

File: .github/workflows/update-deps.yml (hunk 1 above)
Comment (skillitzimberg) [reply .github/workflows/update-deps.yml]: I went ahead and added the `--dev` flag just so we can skip another PR/merge later.

**Pull request merged**